### ISSUE TITLE
Signature mismatch when built with bulk memory + wasm64

### ISSUE THREAD

**Issue by RReverser** [emscripten-core/emscripten#20560]
(Split out from #20549.)

It appears that bumping browser versions in that PR makes BULK_MEMORY enabled by default, as after this bump it's now supported by all engines.

That, in turn, breaks wasm64 tests because bulk memory seems incompatible with wasm64 at the moment, or rather `emscripten_memset_bulkmem` is due to wrong signature:

```
+RuntimeError: Aborted(CompileError: WebAssembly.instantiate(): Compiling function #15:"emscripten_memset_bulkmem" failed: memory.fill[2] expected type i32, found local.get of type i64 @+1859)
+    at abort (/tmp/emtest_fsexcy95/emscripten_test_other_8gw_9wy3/a.out.js:666:11)
+    at /tmp/emtest_fsexcy95/emscripten_test_other_8gw_9wy3/a.out.js:780:5
+Thrown at:
+    at abort (/tmp/emtest_fsexcy95/emscripten_test_other_8gw_9wy3/a.out.js:666:11)
+    at /tmp/emtest_fsexcy95/emscripten_test_other_8gw_9wy3/a.out.js:780:5
```

This should be fairly easy to reproduce in isolation.

**Comment by RReverser** [comment]
Hm or maybe it is a Node.js issue? I can reproduce it with Node.js 16.2 that comes with emsdk and is used by test runner, but not with Node.js 20.0 I have installed globally.

**Comment by RReverser** [comment]
Since this works with Node 20 and wasm64 is still in development, I'm probably going to just bump wasm64 tests to require Node canary (aka 20+).

**Comment by RReverser** [comment]
Trying to auto-disable bulk memory for wasm64 if targeting old Node: https://github.com/emscripten-core/emscripten/pull/20549/commits/a3e86b916aece50bb31b3a12bdf50d03b5ba7c6f

**Comment by kleisauke** [comment]
It looks like support for this landed in V8 10.0.71 with commit https://github.com/v8/v8/commit/18469ec4bfb301991158d0423f07d53c338d09da, so Node.js 18 (LTS) would be the minimum supported runtime version for this setting combo.
https://nodejs.org/en/blog/announcements/v18-release-announce#v8-101

**Comment by RReverser** [comment]
Nice find!
